TensorFlow einsum behavior change

# Description

In the test suite the `test_einsum[tensorflow]` [test is failing](https://travis-ci.org/diana-hep/pyhf/jobs/548493214#L689-L714) for `tensorflow` `v1.14.0`.

# Expected Behavior

`test_einsum[tensorflow]` passes

# Actual Behavior

```
backend = (<pyhf.tensor.tensorflow_backend.tensorflow_backend object at 0x7f11de50be10>, None)
    def test_einsum(backend):
        tb = pyhf.tensorlib
        x = np.arange(20).reshape(5, 4).tolist()
    
        if isinstance(pyhf.tensorlib, pyhf.tensor.mxnet_backend):
            with pytest.raises(NotImplementedError):
                assert tb.einsum('ij->ji', [1, 2, 3])
        else:
>           assert np.all(tb.tolist(tb.einsum('ij->ji', x)) == np.asarray(x).T.tolist())
tests/test_tensor.py:200: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pyhf/tensor/tensorflow_backend.py:260: in einsum
    return tf.einsum(subscripts, *operands)
../../../virtualenv/python3.6.3/lib/python3.6/site-packages/tensorflow/python/ops/special_math_ops.py:255: in einsum
    input_shapes = [x.get_shape() for x in inputs]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.0 = <list_iterator object at 0x7f11cc06bb38>
>   input_shapes = [x.get_shape() for x in inputs]
E   AttributeError: 'list' object has no attribute 'get_shape'
```

# Steps to Reproduce

Run the test suite.

```
pytest -s tests/test_tensor.py
```

# Checklist

- [x] Run `git fetch` to get the most up to date version of `master`
- [x] Searched through existing Issues to confirm this is not a duplicate issue
- [x] Filled out the Description, Expected Behavior, Actual Behavior, and Steps to Reproduce sections above or have edited/removed them in a way that fully describes the issue

--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@ with open(path.join(this_directory, 'README.md'), encoding='utf-8') as readme_md
 
 extras_require = {
     'tensorflow': [
-        'tensorflow~=1.13',
+        'tensorflow~=1.14',
         'tensorflow-probability~=0.5',
         'numpy<=1.14.5,>=1.14.0',  # Lower of 1.14.0 instead of 1.13.3 to ensure doctest pass
         'setuptools<=39.1.0',

--- a/tests/test_tensor.py
+++ b/tests/test_tensor.py
@@ -197,7 +197,9 @@ def test_einsum(backend):
         with pytest.raises(NotImplementedError):
             assert tb.einsum('ij->ji', [1, 2, 3])
     else:
-        assert np.all(tb.tolist(tb.einsum('ij->ji', x)) == np.asarray(x).T.tolist())
+        assert np.all(
+            tb.tolist(tb.einsum('ij->ji', tb.astensor(x))) == np.asarray(x).T.tolist()
+        )
         assert (
             tb.tolist(
                 tb.einsum('i,j->ij', tb.astensor([1, 1, 1]), tb.astensor([1, 2, 3]))

